File: guides/micronaut-data-mongodb-synchronous/java/src/main/java/example/micronaut/FruitRepository.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.mongodb.annotation.MongoRepository;
import io.micronaut.data.repository.CrudRepository;

import java.util.List;

@MongoRepository // <1>
public interface FruitRepository extends CrudRepository<Fruit, String> {

    @NonNull
    Iterable<Fruit> findByNameInList(@NonNull List<String> names); // <2>
}

File: guides/micronaut-data-mongodb-synchronous/python/src/example/micronaut/fruit_repository.py
from typing import List

from micronaut.data.mongodb.annotation import MongoRepository
from micronaut.data.repository import CrudRepository

from .fruit import Fruit


@MongoRepository  # <1>
class FruitRepository(CrudRepository[Fruit, str]):
    def findByNameInList(self, names: List[str]) -> List[Fruit]: ...  # <2>
